src/main/java/cloud/eppo/ConfigurationStore.java

```diff
@@ -1,53 +1,105 @@
 import org.slf4j.LoggerFactory;
 
 public class ConfigurationStore {
   private static final Logger log = LoggerFactory.getLogger(ConfigurationStore.class);
   private final ObjectMapper mapper = new ObjectMapper().registerModule(EppoModule.eppoModule());
 
-  private ConcurrentHashMap<String, FlagConfig> flags;
-  private ConcurrentHashMap<String, BanditParameters> banditParameters; // TODO: bandit stuff
-  // TODO: another map of bandit flag variations to handle no action case
-  static ConfigurationStore instance = null;
+  private Map<String, FlagConfig> flags;
+  private Map<String, BanditReference> banditReferences;
+  private Map<String, BanditParameters> banditParameters;
 
   public ConfigurationStore() {
-    // TODO: handle caching
+    flags = new ConcurrentHashMap<>();
+    banditReferences = new ConcurrentHashMap<>();
+    banditParameters = new ConcurrentHashMap<>();
   }
 
-  public void setFlagsFromJsonString(String jsonString) throws JsonProcessingException {
-    FlagConfigResponse config = mapper.readValue(jsonString, FlagConfigResponse.class);
+  public void setFlagsFromJsonString(String jsonString) {
+    FlagConfigResponse config;
+
+    try {
+      config = mapper.readValue(jsonString, FlagConfigResponse.class);
+    } catch (JsonProcessingException e) {
+      log.error("Unable to parse flag configuration response");
+      throw new RuntimeException(e);
+    }
+
     if (config == null || config.getFlags() == null) {
       log.warn("Flags missing in configuration response");
       flags = new ConcurrentHashMap<>();
+      banditReferences = new ConcurrentHashMap<>();
     } else {
       // TODO: atomic flags to prevent clobbering like android does
       // Record that flags were set from a response so we don't later clobber them with a
       // slow cache read
       flags = new ConcurrentHashMap<>(config.getFlags());
-      log.debug("Loaded " + flags.size() + " flags from configuration response");
+      banditReferences = new ConcurrentHashMap<>(config.getBanditReferences());
+      log.debug("Loaded {} flags from configuration response", flags.size());
     }
   }
 
-  public BanditParameters getBanditParameters(String banditKey) {
-    return this.banditParameters.get(banditKey);
-  }
-
   public FlagConfig getFlag(String flagKey) {
     if (flags == null) {
-      log.warn("Request for flag " + flagKey + " before flags have been loaded");
+      log.warn("Request for flag {} before flags have been loaded", flagKey);
       return null;
     } else if (flags.isEmpty()) {
-      log.warn("Request for flag " + flagKey + " with empty flags");
+      log.warn("Request for flag {} with empty flags", flagKey);
     }
     return flags.get(flagKey);
   }
+
+  public String banditKeyForVariation(String flagKey, String variationValue) {
+    // Note: In practice this double loop should be quite quick as the number of bandits and bandit
+    // variations will be small. Should this ever change, we can optimize things.
```

Comment:
Agreed - no need to prematurely optimize this.

Reply:
[😅 ](https://github.com/Eppo-exp/php-sdk/pull/35/files#diff-8b2fba0ae27745d3578b6cd1a5f13a6edef0672eeaca2404b159d93b4b7cc82dR52)